account-kit/infra/src/middleware/feeEstimator.ts

```diff
@@ -36,14 +38,23 @@ export const alchemyFeeEstimator: (
   async (struct, { overrides, feeOptions, client: client_ }) => {
     const client = clientHeaderTrack(client_, "alchemyFeeEstimator");
     const transport_ = transport({ chain: client.chain });
-    let [block, maxPriorityFeePerGasEstimate] = await Promise.all([
-      client.getBlock({ blockTag: "latest" }),
-      // it is a fair assumption that if someone is using this Alchemy Middleware, then they are using Alchemy RPC
-      transport_.request({
-        method: "rundler_maxPriorityFeePerGas",
-        params: [],
-      }),
-    ]);
+    const [sender, nonce] = await Promise.all([struct.sender, struct.nonce]);
+    const nonceHex = toHex(nonce);
+
+    const [block, maxPriorityFeePerGasEstimate, existingUserOp] =
+      await Promise.all([
+        client.getBlock({ blockTag: "latest" }),
+        // it is a fair assumption that if someone is using this Alchemy Middleware, then they are using Alchemy RPC
+        transport_.request({
+          method: "rundler_maxPriorityFeePerGas",
+          params: [],
+        }),
+        transport_.request({
+          method: "rundler_getPendingUserOperationBySenderNonce",
+          // @ts-expect-error ... I could not tell you why the fuck this isn't resolving correctly
+          params: [sender, nonceHex],
```

Comment:
the issue here is that `UserOperationStruct_v7` (and v6) type the `sender` as a `string`.

```suggestion
          params: [sender as Address, nonceHex],
```

Reply:
omg thank you